ouster-ros/src/os_sensor_node.cpp

```diff
@@ -90,21 +90,8 @@ class OusterSensor : public OusterSensorNodeBase {
         LifecycleNode::on_activate(state);
         lidar_packet_pub->on_activate();
         imu_packet_pub->on_activate();
-
         allocate_buffers();
-        if (!connection_loop_timer) {
-            // TOOD: replace with a thread instead?
-            connection_loop_timer =
-                rclcpp::create_timer(this, get_clock(), 0s, [this]() {
-                    if (reset_in_progress) return;
-                    auto& pf = sensor::get_format(info);
-                    connection_loop(*sensor_client, pf);
-                });
-        } else {
-            connection_loop_timer->reset();
-        }
-
-        reset_in_progress = false;
+        start_sensor_connection_thread();
```

Comment:
I presume it's not possible to reach this line couple of times and `on_activte()` can happen only once, so we don't need to worry about checking do we have any running thread already?

Reply:
We shouldn't worry about it. These are state managed .. the `on_activate` could only trigger if are at an inactivated state.

Reply:
Good good!